### PR TITLE
feat: node compatibility test

### PR DESCRIPTION
**Pull request by bombguy**
## Problem
We should be ensuring that our repo is able to install, build, and test using the specified node versions.

**Issue number, if available:**

## Changes
- Update `setup_node` action to take in `node-version` input
- Add node versions to test matrix in following jobs: `install`, `build`, `test_with_coverage`
- Add node versions to test matrix for `e2e_deployment`, `e2e_create_amplify`
- Skip `global.crypto` workaround for node > 20 as it is [no longer needed](https://github.com/nodejs/node/pull/42083).

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .github/actions/setup_node/action.yml [hunk 1]
```diff
@@ -1,10 +1,14 @@
 # basic setup that applies to most jobs
 # checks out the repo and configures node
 name: setup_node
+inputs:
+  node-version:
+    description: node version used to configure environment with
+    default: lts/hydrogen
```
Comment (edwardfoyle) [inline .github/actions/setup_node/action.yml:7]: can we use the node version numbers instead?

File: .github/workflows/health_checks.yml [hunk 2]
```diff
@@ -21,30 +21,43 @@ jobs:
...
         os: [ubuntu-latest, macos-14, windows-latest]
+        node: [lts/hydrogen, lts/iron]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 10
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
       - uses: ./.github/actions/setup_node
+        with:
+          node-version: ${{ matrix.node }}
       - uses: ./.github/actions/install_with_cache
```
Comment (edwardfoyle) [inline .github/workflows/health_checks.yml:32]: does the install cache key need to include node version now?

File: .github/workflows/health_checks.yml [hunk 3]
```diff
@@ -21,30 +21,43 @@ jobs:
...
       matrix:
         os: [ubuntu-latest, macos-14, windows-latest]
+        node: [lts/hydrogen, lts/iron]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # version 3.6.0
       - uses: ./.github/actions/setup_node
+        with:
+          node-version: ${{ matrix.node }}
+      - uses: ./.github/actions/install_with_cache
```
Comment (edwardfoyle) [inline .github/workflows/health_checks.yml:60]: restore_build_cache already restores the install cache

File: .github/actions/setup_node/action.yml (hunk 1 above)
Comment (bombguy) [reply .github/actions/setup_node/action.yml:7]: they should both resolve to latest versions. I can update it to numbers

File: .github/workflows/health_checks.yml (hunk 3 above)
Comment (edwardfoyle) [reply .github/workflows/health_checks.yml:63]: Is there a reason we need both install_with_cache and restore_build_cache now?

**Review by sobolk — APPROVED** [review]

File: .github/workflows/health_checks.yml (hunk 3 above)
Comment (bombguy) [reply .github/workflows/health_checks.yml]: The tests were failing earlier for missing node_modules, but it looks like its not needed anymore. Removed install_with_cache here.

**Review by sobolk — APPROVED** [review]

**Comment by bombguy** [comment]
The checks need to be updated with the new names that includes the matrix configurations post merge

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**